iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/payload/evolvable/request/PipeTransferHandshakeV2Req.java

```diff
@@ -0,0 +1,109 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.pipe.connector.payload.evolvable.request;
+
+import org.apache.iotdb.commons.pipe.connector.payload.request.IoTDBConnectorRequestVersion;
+import org.apache.iotdb.commons.pipe.connector.payload.request.PipeRequestType;
+import org.apache.iotdb.service.rpc.thrift.TPipeTransferReq;
+import org.apache.iotdb.tsfile.utils.PublicBAOS;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.thrift.TException;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class PipeTransferHandshakeV2Req extends TPipeTransferReq {
+  private transient Map<String, String> params;
+
+  private PipeTransferHandshakeV2Req() {
+    // Empty constructor
+  }
+
+  public Map<String, String> getParams() {
+    return params;
+  }
+
+  /////////////////////////////// Thrift ///////////////////////////////
+
+  public static PipeTransferHandshakeV2Req toTPipeTransferReq(HashMap<String, String> params)
+      throws TException {
+    final PipeTransferHandshakeV2Req handshakeReq = new PipeTransferHandshakeV2Req();
+
+    handshakeReq.version = IoTDBConnectorRequestVersion.VERSION_1.getVersion();
+    handshakeReq.type = PipeRequestType.HANDSHAKE_V2.getType();
+    handshakeReq.body = ByteBuffer.wrap(SerializationUtils.serialize(params));
+
+    handshakeReq.params = params;
+
+    return handshakeReq;
+  }
+
+  public static PipeTransferHandshakeV2Req fromTPipeTransferReq(TPipeTransferReq transferReq) {
+    final PipeTransferHandshakeV2Req handshakeReq = new PipeTransferHandshakeV2Req();
+
+    handshakeReq.params = SerializationUtils.deserialize(transferReq.body.array());
+
+    handshakeReq.version = transferReq.version;
+    handshakeReq.type = transferReq.type;
+    handshakeReq.body = transferReq.body;
+
+    return handshakeReq;
+  }
+
+  /////////////////////////////// Air Gap ///////////////////////////////
+
+  public static byte[] toTransferHandshakeBytes(HashMap<String, String> params) throws IOException {
+    try (final PublicBAOS byteArrayOutputStream = new PublicBAOS();
+        final DataOutputStream outputStream = new DataOutputStream(byteArrayOutputStream)) {
+      ReadWriteIOUtils.write(IoTDBConnectorRequestVersion.VERSION_1.getVersion(), outputStream);
+      ReadWriteIOUtils.write(PipeRequestType.HANDSHAKE_V2.getType(), outputStream);
+      ReadWriteIOUtils.write(params, outputStream);
```

Comment:
ReadWriteIOUtils may not be compatible with SerializationUtils since the latter one use the "Serialzable" api from java...

Reply:
Consider adding a UT for air gap in handshakeV2, and other reqs may have to add one afterwards..

Reply:
Got it.